master/security/calico-network-policy.md

```diff
@@ -0,0 +1,269 @@
+
+#### Namespaced and global network policies
+**Calico network policy** is a namespaced resource that applies to pods/containers/VMs in that namespace. 
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: <b>NetworkPolicy</b>
+metadata:
+  <b>name</b>: allow-tcp-6379
+  <b>namespace</b>: production
+</pre>
+{: .no-select-button}
+
+**Calico global network policy** is a non-namespaced resource and can be applied to any kind of endpoint (pods, VMs, host interfaces) independent of namespace. 
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: <b>GlobalNetworkPolicy</b>
+metadata:
+  <b>name</b>: allow-tcp-port-6379
+</pre>
+{: .no-select-button}
+
+**Calico network and global policies** are applied using calicoctl. Syntax is similar to Kubernetes, but there a few differences. For help, see [calicoctl user reference]({{site.baseurl}}/{{page.version}}/reference/calicoctl/).
+
+#### Ingress and egress
+Each network policy rule applies to either **ingress** or **egress** traffic. From the point of view of an endpoint (pod, VM, host interface), **ingress** is incoming traffic to the endpoint, and **egress** is outgoing traffic from the endpoint. In a Calico network policy, you create ingress and egress rules independently (egress, ingress, or both). 
+
+You can specify whether policy applies to ingress, egress, or both using the **types** field. If you do not use the types field, Calico defaults to the following values.
+
+
+> | Ingress rule present? | Engress rule present? |      Value      |
+> | :-------------------: | :-------------------: | :-------------: |
+> |          No           |          No           |     Ingress     |
+> |          Yes          |          No           |     Ingress     |
+> |          No           |          Yes          |     Egress      |
+> |          Yes          |          Yes          | Ingress, Egress |
+
+
+#### Network traffic behaviors: deny and allow
+
+The Kubernetes network policy specification defines the following behavior:
+- If no network policies apply to a pod, then all traffic to/from that pod is allowed.
+- If one or more network policies apply to a pod containing ingress rules, then only the ingress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+- If one or more network policies apply to a pod containing egress rules, then only the egress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+
+For compatibility with Kubernetes, **Calico network policy** follows the same behavior for Kubernetes pods.  For other endpoint types (VMs, host interfaces), Calico network policy is default deny. That is, only traffic specifically allowed by network policy is allowed, even if no network policies apply to the endpoint.
+
+### Before you begin...
+None
```

Comment:
I find these sections with `None` to be a bit jarring. Can we just remove the `Before you begin...` heading if there's nothing in it?

Reply:
This is Alex decision for the template. His reasoning is that when/if we ever get the ability to allow customers to comment inline, this invites "why is this blank, or I think there are requirements before you start."  I see your point, but for now, this is the decision. The how to template is a work in progress so if we get other inputs we will change it.